### PR TITLE
Update to c2c@1.8.0

### LINKED ISSUE #979: Some Klee spline shapes do not match C2C plotter.
I was making a new dataset using C2C format and noticed that for some contours that use sides made of RZ splines, the shapes output by Klee do not match the c2c_plotter program. The Klee version of the curved edge is very wavy compared to the C2C version. Here is such a shape:

Maybe it is a difference in how the splines are sampled parametrically?

black3.contour:
```
piece = line(start=(3cm,143cm), end=(4cm,149cm))
piece = rz(units=cm,
  rz=4 149
     17 151
     33 155
     43 155
     46 153
)

piece = line(end=(44cm,146cm))
piece = line(end=(38cm,147cm))
piece = rz(units=cm,
  rz=38 147
     33 147
     26 146
)

piece = line(end=(21cm,144cm))
piece = line(end=(3cm,143cm))
```

black3.yaml (needed for klee):
```
dimensions: 2

shapes:
- name: black3
  material: black
  geometry:
    format: c2c
    path: black3.contour
```

Plotting file using C2C:
```
/usr/gapps/c2c/latest/bin/c2c_plotter black3.contour
```

shaping in klee (with sampling):
```
data_collection_util --min 140 0 --max 170 50 --res 150 200 -p 1
../examples/quest_shaping_driver_ex --mesh-file box_2d.root --shape-file black3.yaml --method 0 -v -o 0 -p omp

```

### LINKED ISSUE #980: Octree failure for C2C contour.
I was making a new C2C dataset and noticed that about a third of my C2C contours caused the Octree in Axom to crash. Many of these contours contain spline edges.

black65.contour
```
piece = line(start=(32cm,128cm), end=(38cm,147cm))
piece = line(end=(44cm,146cm))
piece = rz(units=cm,
  rz=44 146
     43 142
     36 125
     33 119
     30 117
     27 121
)

piece = line(end=(32cm,128cm))
```

black65.yaml
```
dimensions: 2
shapes:
- name: black65
  material: black
  geometry:
    format: c2c
    path: black65.contour
```

Klee command lines (note paths are different here for real files):
```
rzgenie7{whitlocb}36: data_collection_util --min 100 25 --max 150 50 --res 150 200 -p 1
[INFO]: Creating a box mesh of resolution (150,200) and bounding box { min:(100,25); max:(150,50); range:<50,25> }
[INFO]: After transformations, mesh has 30000 elements and (approximate) bounding box { min:(100,25); max:(150,50); range:<50,25> }
[INFO]: Saving mesh file 'box_2d'
rzgenie7{whitlocb}37: ../examples/quest_shaping_driver_ex --mesh-file box_2d.root --shape-file black65.yaml --method 0 -v -o 0 -p omp
[INFO]: After loading mesh has 30000 elements and (approximate) bounding box { min:(100,25); max:(150,50); range:<50,25> }
[INFO]: ========================Sampling InOut fields for shapes========================
[INFO]: --------------------------- Loading shape 'black65' ----------------------------
[INFO]: Reading file: ../../data/shaping/../contours/heroic_roses/black65.contour...
[INFO]: Loading contour with 4 pieces
[INFO]: ---------------------------- Generating the octree -----------------------------
[INFO]: Mesh bounding box: { min:(117.004,27); max:(173.182,52.2397); range:<56.1773,25.2397> }
[INFO]:   Generating InOutOctree over surface mesh with 208 vertices and 200 elements.
[INFO]: 	--Inserting vertices took 0.001122898 seconds.
[INFO]: 	--Updating mesh took 0.00115438 seconds.
[INFO]:   After inserting vertices, reindexed mesh has 200 vertices and 200 cells.
[DEBUG]: Inside InOutOctree::checkValid() to verify state of PR octree
[DEBUG]: --Checking that each vertex is in a leaf block of the tree.
[DEBUG]: done.
[DEBUG]: 	Inserting cells into level 0 took 0.000848126 seconds.
[DEBUG]: 	Inserting cells into level 1 took 0.000841897 seconds.
[DEBUG]: 	Inserting cells into level 2 took 0.000814332 seconds.
[DEBUG]: 	Inserting cells into level 3 took 0.000979541 seconds.
[DEBUG]: 	Inserting cells into level 4 took 0.0011968 seconds.
[DEBUG]: 	Inserting cells into level 5 took 0.00116284 seconds.
[DEBUG]: 	Inserting cells into level 6 took 0.00102629 seconds.
[DEBUG]: 	Inserting cells into level 7 took 0.000425237 seconds.
[DEBUG]: 	Inserting cells into level 8 took 0.000124968 seconds.
[DEBUG]: 	Inserting cells into level 9 took 7.7409e-05 seconds.
[DEBUG]: 	Inserting cells into level 10 took 4.3584e-05 seconds.
[DEBUG]: 	Inserting cells into level 11 took 5.0667e-05 seconds.
[DEBUG]: 	Inserting cells into level 12 took 5.459e-05 seconds.
[DEBUG]: 	Inserting cells into level 13 took 5.3638e-05 seconds.
[DEBUG]: 	Inserting cells into level 14 took 5.3061e-05 seconds.
[DEBUG]: 	Inserting cells into level 15 took 5.2852e-05 seconds.
[DEBUG]: 	Inserting cells into level 16 took 4.5964e-05 seconds.
[DEBUG]: 	Inserting cells into level 17 took 5.5759e-05 seconds.
[DEBUG]: 	Inserting cells into level 18 took 5.5097e-05 seconds.
[DEBUG]: 	Inserting cells into level 19 took 5.2937e-05 seconds.
[DEBUG]: 	Inserting cells into level 20 took 5.2406e-05 seconds.
[DEBUG]: 	Inserting cells into level 21 took 5.2285e-05 seconds.
[DEBUG]: 	Inserting cells into level 22 took 5.3281e-05 seconds.
[DEBUG]: 	Inserting cells into level 23 took 5.4209e-05 seconds.
[DEBUG]: 	Inserting cells into level 24 took 5.271e-05 seconds.
[DEBUG]: 	Inserting cells into level 25 took 5.3548e-05 seconds.
[DEBUG]: 	Inserting cells into level 26 took 5.3548e-05 seconds.
[DEBUG]: 	Inserting cells into level 27 took 5.3823e-05 seconds.
[DEBUG]: 	Inserting cells into level 28 took 5.3194e-05 seconds.
[DEBUG]: 	Inserting cells into level 29 took 5.3634e-05 seconds.
[ERROR]: Failed Assert: idx >= 0 && idx < SetPosition(m_data.size())
Attempted to access element 31 but map's data has size 31
** StackTrace of 12 frames **
Frame 1: axom::slic::logErrorMessage(std::string const&, std::string const&, int)
Frame 2: axom::slam::Map<axom::spin::OctreeLevel<2, axom::quest::InOutBlockData>*, axom::slam::Set<int, int>, axom::slam::policies::STLVectorIndirection<int, axom::spin::OctreeLevel<2, axom::quest::InOutBlockData>*>, axom::slam::policies::StrideOne<int> >::verifyPositionImpl(int) const
Frame 3: axom::slam::Map<axom::spin::OctreeLevel<2, axom::quest::InOutBlockData>*, axom::slam::Set<int, int>, axom::slam::policies::STLVectorIndirection<int, axom::spin::OctreeLevel<2, axom::quest::InOutBlockData>*>, axom::slam::policies::StrideOne<int> >::operator[](int)
Frame 4: axom::spin::OctreeBase<2, axom::quest::InOutBlockData>::getOctreeLevel(int)
Frame 5: axom::spin::OctreeBase<2, axom::quest::InOutBlockData>::refineLeaf(axom::spin::OctreeBase<2, axom::quest::InOutBlockData>::BlockIndex const&)
Frame 6: axom::quest::InOutOctree<2>::insertMeshCells()
Frame 7: axom::quest::InOutOctree<2>::generateIndex()
Frame 8: axom::quest::shaping::InOutSampler<2>::initSpatialIndex(double)
Frame 9: axom::quest::SamplingShaper::prepareShapeQuery(axom::klee::Dimensions, axom::klee::Shape const&)
Frame 10: main
Frame 11: __libc_start_main
Frame 12: ../examples/quest_shaping_driver_ex() [0x974f06]
=====

```
black65

### LINKED ISSUE #1042: Update c2c library to v1.8.0
`c2c@1.8.0` includes a bugfix that should resolve #980 and #979

### PR DESCRIPTION
**Pull request by kennyweiss**
# Summary

- This PR is a feature
- It updates our c2c dependency to c2c@1.8.0
- Resolves #1042 
- Fixes #980 
- Fixes #979 

#### TODO

- [x] Update RZ host-configs
- [x] Update CZ host-configs
- [x] Update RELEASE-NOTES
- [x] Confirm that #979 is resolved
- [x] Confirm that #980 is resolved

### CODE REVIEW

File: src/thirdparty/tests/mfem_smoke.cpp [hunk 1]
```diff
@@ -12,6 +12,19 @@
...
+  std::cout << "Using mfem version: "     //
+            << MFEM_VERSION_MAJOR << "."  //
+            << MFEM_VERSION_MINOR << "."  //
+            << MFEM_VERSION_PATCH << std::endl;
+
+  EXPECT_TRUE(MFEM_VERSION_MAJOR >= 0);
+  EXPECT_TRUE(MFEM_VERSION_MINOR >= 0);
+  EXPECT_TRUE(MFEM_VERSION_PATCH >= 0);
+}
+
```
Comment (kennyweiss) [inline src/thirdparty/tests/mfem_smoke.cpp:27]: I was also planning to update to mfem@4.5.2 but it depends on raja@2022.10.* and we're currently on raja@2022.03.*

The issue is related to mfem's usage of `raja::launch`, which axom is not currently using, so I'd expect the update to be relatively straightforward.

**Comment by kennyweiss** [comment]
@BradWhitlock -- I removed the `ENABLE_WHEN_C2C_IS_UPDATED_ON_ALL_PLATFORMS` guards in the `quest_intersection_shaper` tests.

What else should I look into to confirm that #979 and #980 are resolved?
Do I need to modify the `heroic roses` assemblies in the axom_data repo?

**Review by white238 — APPROVED** [review]

File: scripts/spack/packages/c2c/package.py [hunk 2]
```diff
@@ -10,9 +10,10 @@ class C2c(CMakePackage):
     homepage = 'https://rzlc.llnl.gov/c2c'
     url = 'file:///collab/usr/gapps/c2c/dist/c2c-0.11.0.tgz'
 
-    version('develop', git='ssh://git@rz-bitbucket.llnl.gov:7999/wp/copa.git',
```
Comment (rhornung67) [inline scripts/spack/packages/c2c/package.py:13]: wow. that hasn't changed in a while.

File: src/thirdparty/tests/mfem_smoke.cpp (hunk 1 above)
Comment (rhornung67) [reply src/thirdparty/tests/mfem_smoke.cpp:27]: Make sure to use v2022.10.5. There are incremental changes from 2022.10.0 to 2022.10.5 to fix various build issues. It was an unfortunate time for our lack of comprehensive testing.

**Review by agcapps — APPROVED** [review]

**Review by rhornung67 — APPROVED** [review]

File: src/thirdparty/tests/mfem_smoke.cpp (hunk 1 above)
Comment (kennyweiss) [reply src/thirdparty/tests/mfem_smoke.cpp:27]: I created an issue to track our TPL updates: https://github.com/LLNL/axom/issues/1093

**Pull request merged**